lib/active_merchant/billing/gateways/trans_first_transaction_express.rb

```diff
@@ -408,14 +508,28 @@ def message_from(succeeded, response)
       end
 
       def authorization_from(action, response)
-        authorization = response['tranNr'] || response['pmtId']
+        return unless response
+
+        authorization = response['tranData'].try(:[], 'tranNr') || response['tranNr'] || response['pmtId']
 
         # guard so we don't return something like "purchase|"
         return unless authorization
 
         [action, authorization].join(AUTHORIZATION_FIELD_SEPARATOR)
       end
 
+      def avs_from(response)
+        result = response['authRsp'].try(:[], 'avsRslt') || response['avsRslt']
+
+        AVSResult.new(code: result)
+      end
+
+      def cvv_from(response)
+        result = response['authRsp'].try(:[], 'secRslt') || response['secRslt']
+
+        CVVResult.new(result)
+      end
+
```

Comment:
This is a bit different then: https://github.com/rednovalabs/active_merchant/pull/9/commits/9fb6335bac7f5331f831f88fc19b518b43b1049b